### PR TITLE
Add a new tool "search in document".

### PR DESCRIPTION
**Pull request by albig**
This adds a input field to the toolbox in case a fulltext file is
present. The search is executed only in this document. 20 results are
listed and you may page through all results in pages with 20 items.

The search is inspired by the SRU search in DFG-Viewer.

Please note: there is no design provided for this feature (as usual in
Kitodo.Presentation).

### CODE REVIEW

File: Classes/Common/Solr.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ public static function escapeQuery($query) {
         $helper = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\Solarium\Core\Query\Helper::class);
         // Escape query phrase or term.
         if (preg_match('/^".*"$/', $query)) {
-            return '"'.$helper->escapePhrase(trim($query, '"')).'"';
+            return $helper->escapePhrase(trim($query, '"'));
```
Comment (sebastian-meyer) [inline Classes/Common/Solr.php:113]: Why did you remove the quotes? Without quotes phrases would become simple multi-term searches.

File: Classes/Plugin/Tools/SearchInDocumentTool.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        }
+
+        // Quit if no fulltext file is present
+        $fullTextFile = $this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$this->conf['fileGrpFulltext']];
+        if (empty($fullTextFile)) {
+            return $content;
+        }
+
+        // Load template file.
+        $this->getTemplate();
```
Comment (sebastian-meyer) [inline Classes/Plugin/Tools/SearchInDocumentTool.php]: Template loading for tools has changed slightly (see #392).

```suggestion
        $this->getTemplate('###TEMPLATE###', '', TRUE);
```

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

File: Classes/Common/Solr.php (hunk 1 above)
Comment (sebastian-meyer) [reply Classes/Common/Solr.php:113]: This affects every search, not only 'Search in Document'.

Comment (albig) [reply Classes/Common/Solr.php:113]: This should have been another pull-request as this is a bug in Kitodo.Presentation since the switch to Solarium. At the moment, Kitodo.Presentation can't do phrase searches :-(

Example:

current patch
-----------------
The $query arrives at line 113 with this content:
`$query=""festung dresden""`
With trim($query, '"') we get
`$query = "festung dresden"`
With Solarium escapePhrase adds the Quotes again and we return at this point:
`return ""festung dresden""`

This results finally in the folowing query in case of the searchInDocument eID-script:
`http://solrserver:8983/solr/dlfCore2/select?wt=json&q=fulltext:(%22festung%20dresden%22)%20AND%20uid:11&hl=on&hl.fl=fulltext&fl=uid,id,page&hl.method=fastVector&start=0&rows=20`

Solr sees the following query parameter and executes a phrase search:
`"q":"fulltext:(\"festung dresden\") AND uid:11",`

Result textsnippet e.g.:
`"fulltext":["17. April 1945 Seite 19 Hermann Rahne Die »<em>Festung Dresden</em>« Seite 32 Christian Hermann Millionen Krie"]},`

current master
-------------------
With the additional quotes, the result would be:
`http://solrserver:8983/solr/dlfCore2/select?wt=json&q=fulltext:(%22%22festung%20dresden%22%22)%20AND%20uid:11&hl=on&hl.fl=fulltext&fl=uid,id,page&hl.method=fastVector&start=0&rows=20`

Solr sees the following query parameter and does a simple term search:
`"q":"fulltext:(\"\"festung dresden\"\") AND uid:11",`

Resulting textsnippet e.g.:
` "fulltext":["19 Hermann Rahne Die »<em>Festung</em> <em>Dresden</em>« von 1945 Am 16. April 1945 konnten die Dresdner Bürger in der"]},`

Comment (sebastian-meyer) [reply Classes/Common/Solr.php:113]: So the problem seems to be: Why is the query double-quoted in the first place?

Comment (albig) [reply Classes/Common/Solr.php:113]: It's not double quoted in the first place. That's only the output of xdebug.

The question is, why add the quotes another time? This is done already by escapePhrase(). This is what is does:
<pre>
public function escapePhrase($input)
{
return '"' . preg_replace('/("|\\\)/', '\\\$1', $input) . '"';
}
</pre>

Comment (sebastian-meyer) [reply Classes/Common/Solr.php:113]: OK, got it!

**Review by sebastian-meyer — APPROVED** [review]
